src/hooks/network/useOnPageIdle.tsx

```diff
@@ -0,0 +1,153 @@
+import { KclManagerEvents } from '@src/lang/KclSingleton'
```

Comment:
Very similar idle logic to the original implementation. The idle events and timeout loop is the same.

In the original implementation and in this new implementation the entire engine connection is teared down.